code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -139,46 +139,30 @@
 	controller = pilot
 	src.client?.color = null // stop being all fucked up and weird aaaagh
 	src.hud?.update_intent()
-	src.update_health_icon()
-	flock.add_control_icon(src, pilot)
+	if (istype(pilot, /mob/living/intangible/flock/flockmind))
+		flock.addAnnotation(src, FLOCK_ANNOTATION_FLOCKMIND_CONTROL)
+	else
+		flock.addAnnotation(src, FLOCK_ANNOTATION_FLOCKTRACE_CONTROL)
 	if (give_alert)
 		boutput(src, "<span class='flocksay'><b>\[SYSTEM: Control of drone [src.real_name] established.\]</b></span>")
 
+///Release control of a drone normally
 /mob/living/critter/flock/drone/proc/release_control(give_alerts = TRUE)
-	src.flock?.hideAnnotations(src)
-	src.is_npc = 1
 	if (give_alerts)
 		emote("beep")
 		say(pick_string("flockmind.txt", "flockdrone_player_kicked"))
-	if(src.client && !controller)
-		// don't know how this happened but you need a controller right now
-		controller = new/mob/living/intangible/flock/trace(src, src.flock)
-	if(controller)
-		if (src.floorrunning)
-			src.end_floorrunning(TRUE)
-		// move controller out
-		controller.set_loc(get_turf(src))
-		// move us over to the controller
-		var/datum/mind/mind = src.mind
-		if (mind)
-			mind.transfer_to(controller)
-		else
-			if (src.client)
-				var/key = src.client.key
-				src.client.mob = controller
-				controller.mind = new /datum/mind()
-				controller.mind.ckey = ckey
-				controller.mind.key = key
-				controller.mind.current = controller
-				ticker.minds += controller.mind
-		flock.remove_control_icon(src)
-		if (give_alerts)
-			flock_speak(null, "Control of drone [src.real_name] surrended.", src.flock)
-		// clear refs
-		controller = null
-		src.update_health_icon()
-
-/mob/living/critter/flock/drone/proc/release_control_abrupt()
+	_drop_control()
+	if (give_alerts)
+		flock_speak(null, "Control of drone [src.real_name] surrended.", src.flock)
```

Comment:
Not necessary, but maybe could just put _drop_control() first and then have one give_alerts section?